internal/controller/argocdrole_controller.go

```diff
@@ -32,18 +33,20 @@ import (
 )
 
 // blank assignment to verify that RoleReconciler implements reconcile.Reconciler
-var _ reconcile.Reconciler = &RoleReconciler{}
+var _ reconcile.Reconciler = &ArgoCDRoleReconciler{}
 
-// RoleReconciler reconciles a Role object
-type RoleReconciler struct {
+// ArgoCDRoleReconciler reconciles a Role object
+type ArgoCDRoleReconciler struct {
 	client.Client
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 }
 
-// +kubebuilder:rbac:groups=argoproj.io,resources=roles,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=argoproj.io,resources=roles/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups=argoproj.io,resources=roles/finalizers,verbs=update
+// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles/finalizers,verbs=update
+// +kubebuilder:rbac:groups=argoproj.io,resources=argocdrolebindings,verbs=get;list
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=*
```

Comment:
Three small improvements:
- we dont need to create/delete CRs from the operator
- no need to update/path anything outside of the status 
- the configmap permissions a too broad
```suggestion
// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles,verbs=get;list;watch
// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles/status,verbs=get;update;patch
// +kubebuilder:rbac:groups=argoproj.io,resources=argocdroles/finalizers,verbs=update
// +kubebuilder:rbac:groups=argoproj.io,resources=argocdrolebindings,verbs=get;list
// +kubebuilder:rbac:groups=core,resources=configmaps,resourceNames=argocd-rbac-cm,verbs=get;list;watch;update;patch
```

Something similar to this should be applied on the rolebinding controller to respect the least-privileges principle

Reply:
@sbeginCoveo will implement these changes, thank you :)